### PR TITLE
Fix credential leakage in go.d/upsd command errors

### PR DESCRIPTION
**Pull request by stelfrag**
##### Summary
- Report only the command verb in upsd command errors, preventing USERNAME and PASSWORD arguments from appearing in logs.
- Treat empty responses as connection errors so callers drop the connection and avoid indexing an empty response.
- Add regression tests for password omission from authentication errors and empty-response error classification.
- Closes CodeQL code scanning alerts go/clear-text-logging 3629 and 3630.

## Summary by CodeRabbit

- **Bug Fixes**
  - Credentials are now excluded from UPSD authentication error messages, reducing the risk of passwords or usernames appearing in logs.
  - Non-sensitive command arguments remain available in errors to support troubleshooting.
  - Empty UPSD responses are reported distinctly from protocol-level command failures.
- **Tests**
  - Added coverage for credential redaction, preservation of diagnostic command details, and empty-response error handling.

### CODE REVIEW

File: src/go/plugin/go.d/collector/upsd/client.go [hunk 1]
```diff
@@ -147,22 +147,33 @@ func (c *upsdClient) sendCommand(cmd string) ([]string, error) {
 		return nil, err
 	}
 	if errMsg != "" {
-		return nil, fmt.Errorf("%w: %s (cmd: '%s')", errUpsdCommand, errMsg, cmd)
+		return nil, fmt.Errorf("%w: %s (cmd: '%s')", errUpsdCommand, errMsg, commandName(cmd))
```
Comment (ilyam8) [inline src/go/plugin/go.d/collector/upsd/client.go]: I don't think this will do because some commands are multi-word, e.g. `LIST UPS` and `LIST VAR`. `commandName()` reduces both to just `LIST`. The list of available commands is in the same file, so this is hard to miss.


@stelfrag I wonder what AI was used for this change?

Comment (stelfrag) [reply src/go/plugin/go.d/collector/upsd/client.go]: Agreed, that was wrong. Only USERNAME and PASSWORD carry credentials; LIST UPS and LIST VAR <ups> are safe and their arguments are exactly what makes the error diagnosable.

Replaced commandName() with redactCommand(), which returns just the verb for USERNAME/PASSWORD and the command verbatim otherwise. getEndLine() goes back to its own strings.Cut - it needs the verb for protocol framing, which is a different concern from redaction.

Tests added: username redaction, and that 'LIST VAR ups1' survives intact in the error.

File: src/go/plugin/go.d/collector/upsd/client_test.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
```
Comment (ilyam8) [inline src/go/plugin/go.d/collector/upsd/client_test.go:1]: Use table-driven tests

<details>
<summary> example </summary>

```go
func TestUpsdClient_commandErrorsDoNotLeakCredentials(t *testing.T) {
	tests := map[string]struct {
		command       string
		responses     map[string][]string
		wantUpsdError bool
		wantContains  string
		wantNotContain string
	}{
		"username protocol error is redacted": {
			command: fmt.Sprintf(commandUsername, testUsername),
			responses: map[string][]string{
				"USERNAME": {"ERR ACCESS-DENIED"},
			},
			wantUpsdError:  true,
			wantContains:   "USERNAME",
			wantNotContain: testUsername,
		},
		"password protocol error is redacted": {
			command: fmt.Sprintf(commandPassword, testPassword),
			responses: map[string][]string{
				"PASSWORD": {"ERR ACCESS-DENIED"},
			},
			wantUpsdError:  true,
			wantContains:   "PASSWORD",
			wantNotContain: testPassword,
		},
		"username empty response is redacted": {
			command:        fmt.Sprintf(commandUsername, testUsername),
			responses:      map[string][]string{},
			wantContains:   "USERNAME",
			wantNotContain: testUsername,
		},
		"password empty response is redacted": {
			command:        fmt.Sprintf(commandPassword, testPassword),
			responses:      map[string][]string{},
			wantContains:   "PASSWORD",
			wantNotContain: testPassword,
		},
		"list ups protocol error is unchanged": {
			command: commandListUPS,
			responses: map[string][]string{
				"LIST": {"ERR UNKNOWN-COMMAND"},
			},
			wantUpsdError: true,
			wantContains:  commandListUPS,
		},
		"list var protocol error keeps ups name": {
			command: fmt.Sprintf(commandListVar, "ups1"),
			responses: map[string][]string{
				"LIST": {"ERR VAR-NOT-SUPPORTED"},
			},
			wantUpsdError: true,
			wantContains:  "LIST VAR ups1",
		},
		"list ups empty response is unchanged": {
			command:      commandListUPS,
			responses:    map[string][]string{},
			wantContains: commandListUPS,
		},
		"logout empty response is unchanged": {
			command:      commandLogout,
			responses:    map[string][]string{},
			wantContains: commandLogout,
		},
	}

	for name, tt := range tests {
		t.Run(name, func(t *testing.T) {
			client := &upsdClient{conn: &fakeSocket{responses: tt.responses}}

			_, err := client.sendCommand(tt.command)

			require.Error(t, err)
			assert.Equal(t, tt.wantUpsdError, errors.Is(err, errUpsdCommand))
			assert.Contains(t, err.Error(), tt.wantContains)

			if tt.wantNotContain != "" {
				assert.NotContains(t, err.Error(), tt.wantNotContain)
			}
		})
	}
}
```
</details>

**Review by ilyam8 — APPROVED** [review]
